src/Duo.Clients.Wpf/Messaging/Handlers/ApriManutenzioneProdottoMessageHandler.cs

```diff
@@ -0,0 +1,34 @@
+﻿
+using Duo.Clients.Wpf.Presentation;
+using System.Windows;
+using Topics.Radical.ComponentModel.Messaging;
+using Topics.Radical.Messaging;
+using Topics.Radical.Windows.Presentation.ComponentModel;
+
+namespace Duo.Clients.Wpf.Messaging.Handlers
+{
+    class ApriManutenzioneProdottoMessageHandler : AbstractMessageHandler<ApriManutenzioneProdottoMessage>, INeedSafeSubscription
+    {
+        readonly IViewResolver viewResolver;
+        readonly IConventionsHandler conventions;
+
+        public ApriManutenzioneProdottoMessageHandler(IViewResolver viewResolver, IConventionsHandler conventions)
+        {
+            this.viewResolver = viewResolver;
+            this.conventions = conventions;
+        }
+        public override void Handle(object sender, ApriManutenzioneProdottoMessage message)
+        {
+            var view = viewResolver.GetView<ManutenzioneProdottoView, ManutenzioneProdottoViewModel>(vm =>
+                                                                                                        {
+                                                                                                            vm.Id = message.Id;
+                                                                                                            vm.Version = message.Version;
+                                                                                                            vm.Codice = message.Codice;
+                                                                                                            vm.Descrizione = message.Descrizione;
+                                                                                                            vm.Spessore = message.Spessore;
+                                                                                                        });
```

Comment:
Concordo! :)

 

Da: Mauro Servienti [mailto:notifications@github.com] 
Inviato: 31 March 2016 11:33
A: DuoMes/duo <duo@noreply.github.com>
Cc: Nazareno Manco <nazareno.m@live.com>
Oggetto: Re: [DuoMes/duo] Aggiunta gestione anagrafica prodotti (#60)

 

In src/Duo.Clients.Wpf/Messaging/Handlers/ApriManutenzioneProdottoMessageHandler.cs <https://github.com/DuoMes/duo/pull/60#discussion_r58025177> :

> +
> +        public ApriManutenzioneProdottoMessageHandler(IViewResolver viewResolver, IConventionsHandler conventions)
> +        {
> +            this.viewResolver = viewResolver;
> +            this.conventions = conventions;
> +        }
> +        public override void Handle(object sender, ApriManutenzioneProdottoMessage message)
> +        {
> +            var view = viewResolver.GetView<ManutenzioneProdottoView, ManutenzioneProdottoViewModel>(vm =>
> +                                                                                                        {
> +                                                                                                            vm.Id = message.Id;
> +                                                                                                            vm.Version = message.Version;
> +                                                                                                            vm.Codice = message.Codice;
> +                                                                                                            vm.Descrizione = message.Descrizione;
> +                                                                                                            vm.Spessore = message.Spessore;
> +                                                                                                        });

In questo caso invece che passare i dati da modificare con il messaggio io passerei il solo Id e farei si che il VM recuperi dal server la View ad hoc che gli serve per visualizzare la cosa da modificare. In questo caso stai partendo dal presupposto che la view usata per visualizzare il dato contenga tutto quello che serve per fare anche le modifiche. Magari in questo caso va benissimo, facciamoci un ragionamento.

—
You are receiving this because you were assigned.
Reply to this email directly or view it on GitHub <https://github.com/DuoMes/duo/pull/60/files/14b9e83407502201e28f51bc6b6dc3651c9b4a8c#r58025177>   <https://github.com/notifications/beacon/AKnpak5EBCSy8mYkiiDl3tbnKE-rSNNmks5py5TNgaJpZM4H7fBR.gif>